content/guides/supported-idl.md

```diff
@@ -0,0 +1,245 @@
+---
+Title:       What syntax and datatypes are supported in IDL
+Author:      Thijs Miedema
+Date:        2022-08-17
+Template:    plain_markdown
+---
+
+# What syntax and datatypes are supported in IDL
+
+This is a fairly plain description of what is and isn't supported in IDL for CycloneDDS.
+
+[TOC]
+
+# Datatypes
+
+## Primitive types
+
+| Name                            | Description                                                       |
+|---------------------------------|-------------------------------------------------------------------|
+| `int8`                          | Signed 8 bit integer                                              |
+| `int16`, `short`                | Signed 16 bit integer                                             |
+| `int32`, `long`                 | Signed 32 bit integer                                             |
+| `int64`, `long long`            | Signed 64 bit integer                                             |
+| `uint8`                         | Unsigned 8 bit integer                                            |
+| `uint16`, `unsigned short`      | Unsigned 16 bit integer                                           |
+| `uint32`, `unsigned long`       | Unsigned 32 bit integer                                           |
+| `uint64`, `unsigned long long`  | Unsigned 64 bit integer                                           |
+| `char`                          | Single character byte, signed or unsigned depending on platform   |
+| `octet`                         | Single byte                                                       |
+| `boolean`                       | True or false                                                     |
+| `float`                         | 32 bit floating point                                             |
+| `double`                        | 64 bit floating point                                             |
+
+Explicitly **unsupported** are `wchar` and `long double`. We do not bind the `int` type to nudge users to specify how many bits they need.
+
+## Template types
+
+| Name                     | Description                                                                       |
+|--------------------------|-----------------------------------------------------------------------------------|
+| `string`                 | A string without bounds, length is encoded as 32 bit integer, so maximum is 4GB.  |
+| `string<$bound>`         | A string with a maximum of `$bound` characters (excluding null terminator for C). |
+| `sequence<$type> `       | A sequence (indexed container) of `$type` without bounds (still maximum 4GB).     |
+| `sequence<$type,$bound>` | A bounded sequence of `$type` with maximum `$bound` entries.                      |
+
+Explicitly **unsupported** right now are `wstring` and `fixed`. `map` is also unsupported by C and IDL but you can use a `Dict` in Python, which will not inter-operate.
```

Comment:
It is not that it is unsupported by IDL, but rather that it is unsupported by the IDL compiler, right?

```suggestion
Explicitly **unsupported** right now are `wstring` and `fixed`. `map` is also unsupported by C and IDLC but you can use a `Dict` in Python, which will not inter-operate.
```